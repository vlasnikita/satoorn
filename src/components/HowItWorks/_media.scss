@media only screen and (max-width: 1250px) {
    .HowItWorks {
        &__slider {
            width: 1050px;
            height: 514px;
        }
    }
}

@media only screen and (max-width: 1150px) {
    .HowItWorks {
        &__slider {
            width: 940px;
            height: 460px;
        }
    }
}

@media only screen and (max-width: 1050px) {
    .HowItWorks {
        &__slider {
            width: 830px;
            height: 406px;
        }
    }
}

@media only screen and (max-width: 950px) {
    .HowItWorks {
        &__slider {
            width: 750px;
            height: 366px;
        }
    }
}

@media only screen and (max-width: 850px) {
    .HowItWorks {
        &__slider {
            width: 650px;
            height: 316px;
        }
    }
}

@media only screen and (max-width: 750px) {
    .HowItWorks {
        &__slider {
            width: 560px;
            height: 274px;
        }

        &__navigation {
            bottom: 12px;
        }
    }
}

@media only screen and (max-width: 650px) {
    .HowItWorks {
        &__slider {
            width: 100vw;
            height: 48vw;
        }
    }
}

@media only screen and (max-width: 500px) {
    .HowItWorks {
        &__pdf {
            display: flex;
        }

        &__slider {
            display: none;
        }
    }
}
