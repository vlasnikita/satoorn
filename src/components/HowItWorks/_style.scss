.HowItWorks {
    height: calc(100% - 100px);
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    position: relative;
    padding-top: 50px;

    &__pdf {
        display: none;
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        height: 50vh;
        width: 50vh;
        max-height: 50vw;
        max-width: 50vw;
        justify-content: center;
        align-items: center;
        flex-direction: column;
        background: #272939;
        cursor: pointer;
        border-radius: 3px;
        margin: 0 auto;
        box-shadow: 0 1px 5px 0 rgba(60, 60, 60, 0.05);
        text-decoration: none;

        img {
            height: 40%;
            width: auto;
        }

        p{
            margin-top: 10px;
            font-size: 1.8rem;
            color: #eee;
            text-align: center;
        }
    }

    &__slider {
        position: relative;
        background: #fff;
        height: 90%;
        width: auto;
        box-shadow: 0 20px 30px 0px rgba(60, 60, 60, 0.1);
        margin: auto;
    }

    &__slide {
        height: 100%;
    }

    &__navigation {
        position: absolute;
        bottom: 30px;
        left: 50%;
        transform: translateX(-50%);
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
    }

    &__crumbs {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        align-items: center;
        width: 55px;
        height: 5px;
        margin-bottom: 10px;
    }

    &__crumb {
        height: 5px;
        width: 5px;
        background: #eee;
        border-radius: 50%;

        &_active {
            background: $mainPurple;
        }
    }

    &__buttons {
        position: relative;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    &__next {
        display: flex;
        margin: 0 auto;
        min-width: 120px;
        width: auto;
        padding: 0 15px;
    }

    &__prev {
        position: absolute;
        width: 32px;
        right: 100%;

        img {
            height: 12px;
        }
    }
}

@import "/media";