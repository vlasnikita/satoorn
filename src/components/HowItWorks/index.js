import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { withRouter } from "react-router-dom";

import { toggleCommonOnboarding } from 'Fetchers/index'
import { logUserAction } from "Utils/metrics";
import { getSettings } from "Actions/configs";

import arrowIcon from 'Static/arrow.svg'
import slide1 from 'Static/howitworks/howitworks-1.png'
import slide2 from 'Static/howitworks/howitworks-2.png'
import slide3 from 'Static/howitworks/howitworks-3.png'
import slide4 from 'Static/howitworks/howitworks-4.png'
import slide5 from 'Static/howitworks/howitworks-5.png'
import slide6 from 'Static/howitworks/howitworks-6.png'
import pdf from 'Static/how-it-works.pdf'
import fileIcon from 'Static/howitworks/howitworks-icon.svg'

const slideMap = {
    1: slide1,
    2: slide2,
    3: slide3,
    4: slide4,
    5: slide5,
    6: slide6
};

class HowItWorks extends Component {

    state = {
        currentSlide: 1,
        isGoToChallenges: false
    };

    componentDidMount() {
        document.addEventListener('keydown', this.navigateByArrow);

        this.acceptOnboarding()
    }

    componentWillUnmount() {
        document.removeEventListener('keydown', this.navigateByArrow);
    }

    getProfileData() {
        return this.props.profile.profile;
    }

    acceptOnboarding = () => {
        localStorage.setItem('__SATOORN__clientOnboardingConfirmed', 'true');
        logUserAction('ONBOARDING', 'CONFIRMED', this.getProfileData().id);
        if (!!this.getProfileData().id) {
            toggleCommonOnboarding()
                .then(() => {
                    localStorage.removeItem('__SATOORN__clientOnboardingConfirmed');
                    this.props.handleGetSettings()
                })
        }
        else {
            this.forceUpdate()
        }
    };

    navigateByArrow = e => {
        const key = e.key;

        if (key == 'ArrowLeft' && this.state.currentSlide > 1) this.setPrevSlide();
        else if (key == 'ArrowRight' && this.state.currentSlide < 6) this.setNextSlide()
    }

    getSlide = () => {
        return (
            <img
                src={slideMap[this.state.currentSlide]}
                className="HowItWorks__slide"
            />
        )
    };

    getCrumbs = () => {
        return (
            <div className="HowItWorks__crumbs">
                {[1, 2, 3, 4, 5, 6].map(i => (
                    <span key={i} className={`HowItWorks__crumb ${this.state.currentSlide == i ? 'HowItWorks__crumb_active' : ''}`}/>
                ))}
            </div>
        )
    };

    setNextSlide = () => {
        if (this.state.currentSlide < 6) this.setState({ currentSlide: this.state.currentSlide + 1 });
        else this.setState({ isGoToChallenges: true })
    };

    setPrevSlide = () => this.setState({ currentSlide: this.state.currentSlide - 1 });

    render() {
        if (this.state.isGoToChallenges) return <Redirect to={`/challenges`} />;
        else return (
            <div className='HowItWorks'>
                <a
                    target="_blank"
                    href={pdf}
                    className="HowItWorks__pdf"
                >
                    <img src={fileIcon}/>
                    <p>Открыть PDF</p>
                </a>
                <div className="HowItWorks__slider">
                    {this.getSlide()}
                    <div className="HowItWorks__navigation">
                        {this.getCrumbs()}
                        <div className="HowItWorks__buttons">
                            {this.state.currentSlide > 1 && this.state.currentSlide < 6 &&
                            <button
                                onClick={this.setPrevSlide}
                                className="Button Button_white HowItWorks__prev"
                            >
                                <img src={arrowIcon} />
                            </button>
                            }
                            <button
                                onClick={this.setNextSlide}
                                className="Button Button_purple HowItWorks__next"
                            >{this.state.currentSlide == 6 ? 'К топ-челленджам' : 'Далее'}</button>

                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    profile: state.profile,
    settings: state.configs.settings
});

const mapDispatchToProps = (dispatch) => {
    return {
        handleGetSettings: () => dispatch(getSettings())
    }
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(HowItWorks))