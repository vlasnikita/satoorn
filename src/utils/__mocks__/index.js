const utils = jest.requireActual('Utils/index')

export const getOperationHash = jest.fn().mockImplementation(str => 'MOCKED_HASH')

export const getDonatersEmoji = jest.fn().mockImplementation(() => utils.getDonatersEmoji())

export const linkifyPlainText = jest.fn().mockImplementation((text) => utils.linkifyPlainText(text))

export const openPaymentWidget =  jest.fn().mockImplementation((amount, transactionId, desc, isDonation, challengeId, challengeBody) =>{})