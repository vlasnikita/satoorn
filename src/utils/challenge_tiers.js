import venzelGreenIcon from "Static/challenge/venzel-green.svg";
import venzelBlueIcon from "Static/challenge/venzel-blue.svg";
import venzelPurpleIcon from "Static/challenge/venzel-purple.svg";
import venzelYellowIcon from "Static/challenge/venzel-yellow.svg";
import React from "react";

const TIER_LIMITS_RUR = {
    t5: 0,
    t4: 150,
    t3: 300,
    t2: 700,
    t1: 1000
}

export const getTierNumber = donationAmountString => {
    const donationAmount = +donationAmountString

    if(donationAmount < TIER_LIMITS_RUR.t4) return 5
    else if(donationAmount < TIER_LIMITS_RUR.t3) return 4
    else if(donationAmount < TIER_LIMITS_RUR.t2) return 3
    else if(donationAmount < TIER_LIMITS_RUR.t1) return 2
    else return 1
}

export const getVenzel = donationAmountString => {
    const donationAmount = +donationAmountString
    const tier = getTierNumber(donationAmount);

    if (tier === 4) return <img className='Challenge__venzel' src={venzelGreenIcon} alt=""/>;
    else if (tier === 3) return <img className='Challenge__venzel' src={venzelBlueIcon} alt=""/>;
    else if (tier === 2) return <img className='Challenge__venzel' src={venzelPurpleIcon} alt=""/>;
    else if (tier === 1) return <img className='Challenge__venzel' src={venzelYellowIcon} alt=""/>
};