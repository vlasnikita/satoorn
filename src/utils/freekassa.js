import md5 from 'md5'

const merchant_id = 17257
const secret = 'lqEKapouwJ4MUBH'

export function generateFKPaymentLink (sum, challenge_id){
    const url_base = 'https://pay.freekassa.ru/'
    const currency = 'RUB'
    const md5_input = `${merchant_id}:${sum}:${secret}:${currency}:${challenge_id}`
    const signature = md5(md5_input)
    
    const url = `${url_base}?m=${merchant_id}&oa=${sum}&currency=${currency}&o=${challenge_id}&s=${signature}`
    return url
}