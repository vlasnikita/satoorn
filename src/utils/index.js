import {UnitPay} from 'Static/js/unitpay'
import {YooMoneyCheckout} from 'Static/js/yoomoney'
import sha256 from 'crypto-js/sha256'
import QRCode  from 'qrcode';

import store from 'Store/index';
import { SECRET_KEY, PUBLIC_KEY, HOST } from 'Constants/payment_module'
import { HOSTING_ADDRESS } from 'Constants/routes'
import ALERT_TYPE from 'Constants/alert_type'

import { setActiveTransactionSuccess, setActiveTransactionFail } from 'Actions/transactions'
import { setRedirectToChallenge } from 'Actions/challenges'
import { updateChallenge } from 'Actions/challenges'
import { showAlert } from 'Actions/alert'
import { getChallengeById } from 'Actions/challenges';
import log from './logging';

export const validateEmail = email => {
    const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
}

export const validateUrl = url => {
    const re = /(((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:)*@)?(((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?)(:\d*)?)(\/((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)+(\/(([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)*)*)?)?(\?((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|[\uE000-\uF8FF]|\/|\?)*)?(\#((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|\/|\?)*)?$/
    return re.test(String(url).toLowerCase());
}

export const concatenateUrlParams = paramsMap => {
    let urlParamsString = ''
    const isObject = paramsMap === Object(paramsMap) && Object.prototype.toString.call(paramsMap) !== '[object Array]'
    if(!paramsMap || !isObject) return urlParamsString

    const keys = Object.keys(paramsMap)
    if(keys.length === 0) return urlParamsString
    else {
        Object.keys(paramsMap).forEach(paramKey => {
            if(!!paramsMap[paramKey]) {
                if (urlParamsString.length === 0) urlParamsString = `?${paramKey}=${paramsMap[paramKey]}`
                else urlParamsString = `${urlParamsString}&${paramKey}=${paramsMap[paramKey]}`
            }
        })
        return urlParamsString
    }
}

export const getFormSignature = (amount, transaction, desc) => {
    const hashStr = `${transaction}{up}${desc}{up}${amount}{up}${SECRET_KEY}`
    return sha256(hashStr)
}

export const openInNewTab = url => {
    var win = window.open(url, '_blank');
    win.focus();
}

export const openPaymentWidget = (amount, transaction, desc, isDonation, challengeId, challengeBody, streamerLogin) => {
    const signature = getFormSignature(amount, transaction, desc)

    // store.dispatch(getChallengeById(challengeId))

    const checkout = new window.YooMoneyCheckoutWidget({
        confirmation_token: 'confirmation-token', //TODO set proper token
        return_url: 'https://satoorn.ru/', //TODO set redirect 200 url
        customization: {
            //TODO customize view here
            modal: true
        },
        error_callback: function(error) {
            console.log(error)
        }
    });
  
    checkout.on('modal_close', () => {
        console.log('Код, который нужно выполнить после закрытия всплывающего окна.')
    });

    checkout.on('success', () => {
        log.debug('Успешный платеж', params);

        if(isDonation) {
            store.dispatch(updateChallenge(challengeId, {...challengeBody, isJustUpdated: true}))
        }
        else {
            store.dispatch(setActiveTransactionSuccess())
            store.dispatch(setRedirectToChallenge(challengeId, streamerLogin))
        }
        store.dispatch(showAlert(ALERT_TYPE.OK))

        checkout.destroy();
    });

    checkout.on('fail', () => {
        log.debug(message, params);

        store.dispatch(setActiveTransactionFail())
        store.dispatch(showAlert(ALERT_TYPE.PS_ERROR))

        checkout.destroy();
    });

    //Отображение платежной формы в контейнере
    checkout.render('payment-form');


    /*
    ----- OBSOLETE UNITPAY, EXAMPLE PURPOSES -----
    */
    // const payment = new UnitPay();
    // payment.createWidget({
    //     sum: +amount,
    //     desc: desc.toString(),
    //     account: transaction.toString(),
    //     domainName: "unitpay.money",
    //     signature: signature.toString(),
    //     publicKey: PUBLIC_KEY,
    //     locale: 'ru'
    // });

    // payment.success(function (params) {
    //     log.debug('Успешный платеж', params);

    //     if(isDonation) {
    //         store.dispatch(updateChallenge(challengeId, {...challengeBody, isJustUpdated: true}))
    //     }
    //     else {
    //         store.dispatch(setActiveTransactionSuccess())
    //         store.dispatch(setRedirectToChallenge(challengeId, streamerLogin))
    //     }
    //     store.dispatch(showAlert(ALERT_TYPE.OK))
    // });
    // payment.error(function (message, params) {
    //     log.debug(message, params);

    //     store.dispatch(setActiveTransactionFail())
    //     store.dispatch(showAlert(ALERT_TYPE.PS_ERROR))
    // });
    // return false;
}

export const getDonatersEmoji = () => {
    if(process.env.NODE_ENV === 'test') return ''
    const emojis = ['😱', '😵', '🤩', '🤗','🤔', '😮', '😌', '🙃']
    return emojis[ Math.floor(Math.random() * emojis.length) ]
}

export const getAlertSuccessEmoji = () => {
    if(process.env.NODE_ENV === 'test') return ''
    const emojis = [
        '😊','😋','😉','️😏','😌','🤤', '😸','️🙌',
        '👍','👊','✊','✌', '️🤘','🤟','👌','🤙',
        '🐒','🙈', '🌝','🎉','🎊','🔥','✨']

    return emojis[ Math.floor(Math.random() * emojis.length) ]

}

export const getOperationHash = str => {
    return sha256(str).toString()
};

export const linkifyPlainText = text => text.replace(/((http:|https:)[^\s]+[\w])/g, '<a href="$1" target="_blank">$1</a>')

export const checkIfNoOuterUIPage = () => {
    return (
        window.location.pathname === '/policy' ||
        window.location.pathname === '/agreement'
    )
}

export function onlyUnique(value, index, self) {
    return self.indexOf(value) === index;
}

export const getItemByKey = (array, key) => array.reduce((acc, item) => {
    if(!acc) return item.key === key ? item : null
    else return acc
}, null)

export const generateQRCode = async text => {
    try {
      return await QRCode.toDataURL(text)
    } catch (err) {
      console.error(err)
    }
}