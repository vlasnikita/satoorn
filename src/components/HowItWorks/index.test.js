import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import React from 'react'
import 'jest-specific-snapshot'
import HowItWorks from './index'
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {  defaultTestInitialState } from 'Test/utils/commonTestUtils';
import toJson  from 'enzyme-to-json';

configure({ adapter: new Adapter() })
const mockStore = configureMockStore();
jest.mock('Static/how-it-works.pdf', () => ({ /* fake module */ }))

it('renders correctly', () => {
    const tree = mount(
        <BrowserRouter keyLength={0}>
            <Provider store={mockStore(defaultTestInitialState)}>
                <HowItWorks />
            </Provider>
        </BrowserRouter>
    );
    expect(toJson(tree)).toMatchSnapshot();
});