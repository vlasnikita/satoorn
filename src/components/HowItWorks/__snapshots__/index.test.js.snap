// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <withRouter(Connect(HowItWorks))>
        <Route>
          <Connect(HowItWorks)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <HowItWorks
              handleGetSettings={[Function]}
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              profile={
                Object {
                  "loaded": true,
                  "profile": Object {
                    "authenticationSource": "twitch",
                    "bio": null,
                    "challengeCount": 0,
                    "contacts": Array [],
                    "email": null,
                    "externalId": "204865822",
                    "externalPhotoLink": "https://static-cdn.jtvnw.net/user-default-pictures-uv/75305d54-c7cc-40d1-bb9c-91fbe85943c7-profile_image-300x300.png",
                    "externalProfileLink": null,
                    "followerCount": 0,
                    "id": "5f108c57399c8300013dad42",
                    "login": "vlasnikita",
                    "streamer": false,
                    "viewerCount": 0,
                    "walletAmount": 0,
                  },
                }
              }
              settings={
                Object {
                  "loaded": true,
                  "loading": false,
                  "value": Object {
                    "challengeAcceptDetailsConfirmed": true,
                    "challengeDonateDetailsConfirmed": true,
                    "serviceOnboardingConfirmed": true,
                  },
                }
              }
            >
              <div
                className="HowItWorks"
              >
                <a
                  className="HowItWorks__pdf"
                  href={Object {}}
                  target="_blank"
                >
                  <img
                    src="howitworks-icon.svg"
                  />
                  <p>
                    Открыть PDF
                  </p>
                </a>
                <div
                  className="HowItWorks__slider"
                >
                  <img
                    className="HowItWorks__slide"
                    src="howitworks-1.png"
                  />
                  <div
                    className="HowItWorks__navigation"
                  >
                    <div
                      className="HowItWorks__crumbs"
                    >
                      <span
                        className="HowItWorks__crumb HowItWorks__crumb_active"
                        key="1"
                      />
                      <span
                        className="HowItWorks__crumb "
                        key="2"
                      />
                      <span
                        className="HowItWorks__crumb "
                        key="3"
                      />
                      <span
                        className="HowItWorks__crumb "
                        key="4"
                      />
                      <span
                        className="HowItWorks__crumb "
                        key="5"
                      />
                      <span
                        className="HowItWorks__crumb "
                        key="6"
                      />
                    </div>
                    <div
                      className="HowItWorks__buttons"
                    >
                      <button
                        className="Button Button_purple HowItWorks__next"
                        onClick={[Function]}
                      >
                        Далее
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </HowItWorks>
          </Connect(HowItWorks)>
        </Route>
      </withRouter(Connect(HowItWorks))>
    </Provider>
  </Router>
</BrowserRouter>
`;
